### ISSUE TITLE
[49] Open-Vocabulary Image Segmentation (OpenSeg)

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#77]
![image](https://user-images.githubusercontent.com/16400591/147656393-823df64f-0a0a-4403-b778-36ad7b5d1e84.png)
open-segmentation 을 수행하는 논문
[paper](https://arxiv.org/pdf/2112.12143.pdf)

#  Method
방법은 생각보다 간단하다.

1. supervised 모델 훈련
    1. caption 중 명사만 뽑아서 word encoder (pre-trained ALIGN BERT-Large) 돌림
    2. Learning Segmentation masks (아래에서 설명)
    3. visual-semantic alignment (아래에서 설명)
    
2. weakly-supervised 돌리기

![image](https://user-images.githubusercontent.com/16400591/147656412-d0cf7fa6-a20c-4dbc-b5a2-77c4718c6ef6.png)

## Learning Segmentation masks
cross-attention module은,
1. max-deeplab 이나 maskformer 랑 비슷하게 구성
2. 3개 쌓아올렸음

cross-attention 모듈은 그림처럼 class-wise 한 mask를 뽑아냄
query 개수가 N개이기 때문에, mask dimension <img src="https://render.githubusercontent.com/render/math?math=s \in R^{NxHxW}">. 
class agnostic mask annotation <img src="https://render.githubusercontent.com/render/math?math=s^l \in R^{MxHxW}">.

최종 loss 는 M 개 gt 에 대해 matching loss 를 구하는 방식.
![image](https://user-images.githubusercontent.com/16400591/147660521-7fadfbf7-abe3-410e-83c6-d664960d08b8.png)

## visual-semantic alignment
softmax 는 다음과 같이 나타냄 (t 는 learnable temperature)
![image](https://user-images.githubusercontent.com/16400591/147659170-67a8ad47-1aa8-40c4-b0cd-3c7840376f6a.png)

region i 와 word j 의 similarity 는 다음과 같음
![image](https://user-images.githubusercontent.com/16400591/147659491-74cc1f91-dbd5-48ff-bd3a-66b9e5ba2c6a.png)

j번째 word 에 대해서 모든 이미지와의 similarity는 다음과 같음
<img src="https://render.githubusercontent.com/render/math?math=\huge g(z, w_j ) = [<z_1, w_j>, ...,<z_N , w_j>] \in R^{Nx1}">

모든 pair 에 대한 similarity.
b는 batch-index 임
![image](https://user-images.githubusercontent.com/16400591/147659637-f1bafe01-e7be-44b8-acc3-20a12e5b197c.png)

여기서부터 조금 헷갈릴 수 있는데,
batch 안의 모오~든 이미지 I 에 대해서, 특정 batch의 caption (noun sets) 이랑 similarity를 다음과 같이 정의하고
![image](https://user-images.githubusercontent.com/16400591/147659941-6edc48cb-53b8-469f-9478-d144f9fd3530.png)

마찬가지로 batch 안의 모오~든 caption C 에 대해서, 특정 batch의 image의 region들과의 similarity를 다음과 같이 정의한다.
![image](https://user-images.githubusercontent.com/16400591/147660012-a0b65529-f329-4552-a51e-ab29fac9c8a4.png)

최종 loss 는 다음과 같음
![image](https://user-images.githubusercontent.com/16400591/147659704-f146baa0-0ee9-4c49-b66e-80c34d17e138.png)

## Weakly supervision
아 ㅋㅋ 김 빠짐. human annotation이 결국 들어갔음.
segmentation loss L_s 만 써서 먼저 teacher 모델을 학습시키고
[Localized Narrative](https://arxiv.org/pdf/1912.03098.pdf) image-caption 데이터에 inference.
영역별로 class만 human annotation 을 했음.

# Results
Localized Narrative 셋에는 coco, ade20k 등이 다 들어가 있는데, 여기서 ADE20K 는 unseen 으로 놔두고 싶어서 학습 때 제외시킴.
652k training image 사용

![image](https://user-images.githubusercontent.com/16400591/147661762-9fc196e3-49bc-4fbf-ad20-2003a88eeb36.png)
PC == Pascal Context
A == ADE20k
